.github/workflows/update-prerun-script.yml

```diff
@@ -0,0 +1,93 @@
+      - name: Get Version
+        id: get-version
+        run: |
+          current_version=$(.github/script/get-current-version)
+          echo "::set-output name=version-number::$((${current_version%%.*} - 1))"
+
+  update-pre-run-script:
+    name: Update pre run script on Saucelabs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Package
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - run: aws s3 cp s3://saucelabs-prebuild-script/script.sh ./
+      - name: Update pre run script and create PRs
+        run: |
+          RESPONSE="$(curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" —location --request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' --form 'payload=@"script.sh"' --form 'name="blur_replacement_script"')"
+          SCRIPT_ID=$(jq -r '.item.id' <<< "$RESPONSE")
+          echo "Updating integration tests pre-run script ID for origin branch"
+          sed -i -E "s/storage.*'/storage:$SCRIPT_ID'/g" ${GITHUB_WORKSPACE}/integration/js/utils/WebdriverSauceLabs.js
+          rm -rf script.sh
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: Update integration tests pre-run script ID
+          title: Update integration tests pre-run script ID
+          branch: update-pre-run-script-id
+          base: main
+          delete-branch: true
+  
+  update-pre-run-script-previous-version:
+    name: Update pre run script on Saucelabs for previous version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Package
+        uses: actions/checkout@v3
+        with:
+          ref: "release-${{ needs.get-previous-version.outputs.previous_version_number }}.x"
+          fetch-depth: 0
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - run: aws s3 cp s3://saucelabs-prebuild-script/script.sh ./
+      - name: Update pre run script and create PRs
+        run: |
+          RESPONSE="$(curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" —location --request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' --form 'payload=@"script.sh"' --form 'name="blur_replacement_script"')"
+          SCRIPT_ID=$(jq -r '.item.id' <<< "$RESPONSE")
+          echo "Updating integration tests pre-run script ID for origin branch"
+          sed -i -E "s/storage.*'/storage:$SCRIPT_ID'/g" ${GITHUB_WORKSPACE}/integration/js/utils/WebdriverSauceLabs.js
+          rm -rf script.sh
+      - name: Create Pull Request
```

Comment:
Do we really need a github action for this? Can you just use regular GitHub CLI commands?

Reply:
The action gets the credentials directly. I will have to add credentials to GitHub secrets to use the CLI commands

Reply:
which credentials is this using? I dont see how this has the correct permisisons

Reply:
This is using the default `GITHUB_TOKEN` according to [this](https://github.com/marketplace/actions/create-pull-request#action-inputs)

Reply:
Oh right, I rmember now - Github token should be sufficient. We tend to stay away from 3rd party actions as much as possible though. Is it possible to just have a bash script do this with github cli?

Reply:
I did try using GitHub CLI in one of my previous commits. It did not work as it said it did not have enough permissions to create a PR which is why I am using this action now. I will have to create a PAT to add permissions to create a PR

Reply:
If its using the GITHUB_TOKEN, that env variable is available in this context, meaning you should be able to write a script as long as you make sure it has the github token env variable attached to the job that runs the script